pkg/server/spec/container.go

```diff
@@ -36,3 +73,51 @@ func newInitContainers(slugURL, image string, a *app.App, fs storage.Storage) []
 		},
 	}}
 }
+
+func newNginxVolumeMount() *VolumeMounts {
+	return &VolumeMounts{
+		Name:      "nginx-conf",
+		MountPath: nginxConfTmplDir,
+		ReadOnly:  true,
+	}
+}
+
+func newNginxContainer(image string) *Container {
+	args := fmt.Sprintf(nginxArgTmpl, nginxConfTmplDir, nginxConfFile, nginxConfDir, nginxConfFile)
+	port := strconv.Itoa(DefaultPort)
+	backend := fmt.Sprintf(nginxBackendTmpl, secondaryPort)
+
+	return &Container{
+		Name:    "nginx",
+		Image:   image,
+		Command: []string{"/bin/sh"},
+		Args:    []string{"-c", args},
+		Ports: []Port{{
+			Name:          "nginx",
+			ContainerPort: int32(DefaultPort),
+		}},
+		Env: map[string]string{
+			"NGINX_PORT":    port,
+			"NGINX_BACKEND": backend,
+		},
+		VolumeMounts: []*VolumeMounts{
+			newNginxVolumeMount(),
+		},
+	}
+}
+
+func newAppContainer(name, image string, envVars map[string]string, hasNginx bool) *Container {
+	port := DefaultPort
+	if hasNginx {
+		port = secondaryPort
+	}
+	return &Container{
+		Name:  name,
+		Image: image,
+		Env:   envVars,
+		Ports: []Port{{
+			Name:          "app",
+			ContainerPort: int32(port),
+		}},
+	}
+}
```

Comment:
What about a new file for nginx specs? `spec/nginx.go1

Reply:
LGTM, but the nginx is just a new container spec

Reply:
I know... follow your heart ❤️ :shipit: